src/Sentry/Extensibility/SentryStackTraceFactory.cs

```diff
@@ -36,287 +21,18 @@ public class SentryStackTraceFactory : ISentryStackTraceFactory
     /// <returns>A Sentry stack trace.</returns>
     public virtual SentryStackTrace? Create(Exception? exception = null)
     {
-        var isCurrentStackTrace = exception == null && _options.AttachStacktrace;
-
-        if (exception == null && !isCurrentStackTrace)
+        if (exception == null && !_options.AttachStacktrace)
         {
             _options.LogDebug("No Exception and AttachStacktrace is off. No stack trace will be collected.");
             return null;
         }
 
+        var isCurrentStackTrace = exception == null && _options.AttachStacktrace;
         _options.LogDebug("Creating SentryStackTrace. isCurrentStackTrace: {0}.", isCurrentStackTrace);
 
-        return Create(CreateStackTrace(exception), isCurrentStackTrace);
-    }
-
-    /// <summary>
-    /// Creates a s<see cref="StackTrace"/> from the <see cref="Exception"/>.
-    /// </summary>
-    /// <param name="exception">The exception.</param>
-    /// <returns>A StackTrace.</returns>
-    protected virtual StackTrace CreateStackTrace(Exception? exception) =>
```

Comment:
This is a breaking change. Can we get around it? Otherwise we'll need to land this in 4.0.

IIRC this was added to get this package out there: https://www.nuget.org/packages/Tuhu.Sentry.AsyncStackTrace

This is not really needed since Sentry 3.0.0 since we bundled Ben.Demystifier's behaviour already.
So we could also decide to go ahead with the change still on 3.0.0